src/TerritorialCouncil/PoliticalCommitteeManager.php

```diff
@@ -133,6 +159,88 @@ public function handleTerritorialCouncilMembershipUpdate(
+            $this->throwException(
+                'territorial_council.adherent_has_no_membership',
+                [
+                    '{{ email }}' => $adherent->getEmailAddress(),
+                    '{{ territorialCouncil }}' => $territorialCouncil->getNameCodes(),
+                ]
+            );
+        }
+
+        if ($adherent->hasPoliticalCommitteeMembership()) {
+            $this->throwException(
+                'political_committee.membership.adherent_has_already',
+                [
+                    '{{ email }}' => $adherent->getEmailAddress(),
+                    '{{ politicalCommittee }}' => $adherent->getPoliticalCommitteeMembership()->getPoliticalCommittee()->getName(),
+                ]
+            );
+        }
+
+        $politicalCommittee = $territorialCouncil->getPoliticalCommittee();
+        $nbMembers = $this->membershipRepository->countLeaderAndMayorMembersFor($politicalCommittee);
+        if ($nbMembers >= self::MAX_MAYOR_AND_LEADER) {
+            $this->throwException(
+                'political_committee.membership.has_max_number_of_mayor_and_leader',
+                [
+                    '{{ max }}' => self::MAX_MAYOR_AND_LEADER,
+                    '{{ politicalCommittee }}' => $territorialCouncil->getPoliticalCommittee()->getName(),
+                ]
+            );
+        }
+
+        $isMayor = $this->mandateRepository->hasMayorMandate($adherent);
+        $quality = TerritorialCouncilQualityEnum::LEADER;
+        if ($isMayor) {
+            $quality = TerritorialCouncilQualityEnum::MAYOR;
+        }
+
+        $membership = $this->createMembership($adherent, $politicalCommittee, $quality);
+        $adherent->setPoliticalCommitteeMembership($membership);
+
+        $this->entityManager->persist($membership);
+        $this->entityManager->flush();
+    }
+
+    public function removeMayorOrLeaderMembership(TerritorialCouncil $territorialCouncil, Adherent $adherent): void
+    {
+        if (!$adherent->hasTerritorialCouncilMembership()
+            || $adherent->getTerritorialCouncilMembership()->getTerritorialCouncil()->getId() !== $territorialCouncil->getId()) {
+            $this->throwException(
+                'territorial_council.adherent_has_no_membership',
+                [
+                    '{{ email }}' => $adherent->getEmailAddress(),
+                    '{{ territorialCouncil }}' => $territorialCouncil->getNameCodes(),
+                ]
+            );
+        }
+
+        if (!$adherent->hasPoliticalCommitteeMembership()) {
+            $this->throwException(
+                'political_committee.membership.adherent_has_no_membership',
+                [
+                    '{{ email }}' => $adherent->getEmailAddress(),
+                    '{{ territorialCouncil }}' => $territorialCouncil->getNameCodes(),
+                ]
+            );
+        }
```

Comment:
it seems to be the same piece of code for validation, maybe we can move this in a private method?

Reply:
Only a part with first `if` is the same.